product_assortment/models/ir_filters.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models, _
+from odoo.osv import expression
+
+
+class ProductAssortment(models.Model):
+    _inherit = 'ir.filters'
+
+    @api.model
+    def _get_default_model(self):
+        if self.env.context.get('product_assortment', False):
+            model = self.env.ref('product.model_product_product')
+            return model.model
+        return ''
+
+    @api.model
+    def _get_default_is_assortment(self):
+        if self.env.context.get('product_assortment', False):
+            return True
+        return False
+
+    model_id = fields.Selection(
+        default=lambda x: x._get_default_model())
+
+    blacklist_product_ids = fields.Many2many(
+        comodel_name='product.product',
+        relation='assortment_product_blacklisted')
+
+    whitelist_product_ids = fields.Many2many(
+        comodel_name='product.product',
+        relation='assortment_product_whitelisted')
+
+    record_count = fields.Integer(compute='_compute_record_count')
+
+    is_assortment = fields.Boolean(
+        default=lambda x: x._get_default_is_assortment())
+
+    @api.multi
+    def _get_eval_domain(self):
+        res = super(ProductAssortment, self)._get_eval_domain()
+
+        if self.whitelist_product_ids and res:
```

Comment:
@rousseldenis @Cedric-Pigeon In this code whitelist and blacklist seems to be mutually exclusive... but nothing prevents the  user to fill values for both... (A constrains ? or ...)

Reply:
@lmignon what do you think they are mutually exclusive. This is not the case. The goal is to include manually some products which are not selected by the filter and/or exclude product from the filter

Reply:
oups sorry, I misread the code.